### ISSUE TITLE
Expense - Expense report shows a blank message instead of "detached a receipt from expense"

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#44207]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.1.0
**Reproducible in staging?:** y
**Reproducible in production?:** no new feature
**If this was caught during regression testing, add the test name, ID and link from TestRail:** n/a
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to DM.
3. Submit two expenses to the user - one of the expenses must be submitted with a receipt.
4. Go to Troubleshoot > Clear cache and restart > Reset and refresh.
5. Go to the same DM in Step 2.
6. Go to transaction thread of expense with receipt.
7. Click on the receipt.
8. Click 3-dot menu > Delete receipt > Delete the receipt.
9. Go to expense report.

## Expected Result:
"detached a receipt from expense" system message will be visible in the expense report.

## Actual Result:
"detached a receipt from expense" system message is not visible in the expense report.
The message is blank in the report. It only displays the message in LHN preview.
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/38435837/e9fe797f-3bee-449b-b920-8e38db423117

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by m-natarajan** [comment]
@grgia FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by m-natarajan** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by tsa321** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Expense - Expense report shows a blank message instead of "detached a receipt from expense" when user delete receipt.

### What is the root cause of that problem?
When user remove receipt, the server will return  report action with type: `MANAGERDETACHRECEIPT`. and it is categorized as old dot report action:

https://github.com/Expensify/App/blob/4dc1710ea72daaa44875e24ea0f71056b33ed8ef/src/libs/ReportActionsUtils.ts#L1162 

In PR: https://github.com/Expensify/App/pull/40168    `getMessageOfOldDotReportAction` got modified:

https://github.com/Expensify/App/blob/4dc1710ea72daaa44875e24ea0f71056b33ed8ef/src/libs/ReportActionsUtils.ts#L1199-L1203

The content of reportActions is:

![textmessage](https://github.com/Expensify/App/assets/114975543/072e2eec-bb91-47d5-9377-25cfb5bee6a3)

And will make the `getMessageOfOldDotReportAction` give incorrect result.

### What changes do you think we should make in order to solve the problem?
If this is deploy blocker:
We should modify the `getMessageOfOldDotReportAction` into: 
```javascript
function getMessageOfOldDotReportAction(reportAction: OnyxEntry<ReportAction>): string {
    return reportAction?.message?.map((element) => getTextFromHtml(element?.text)).join('') ?? '';
}

**Comment by deetergp** [comment]
I'm pretty sure this is caused [this PR](https://github.com/Expensify/App/pull/40168).

**Comment by tsa321** [comment]
@deetergp yes, the fix is to revert the `getMessageOfOldDotReportAction` into:

function getMessageOfOldDotReportAction(reportAction: OnyxEntry<ReportAction>): string {
    return reportAction?.message?.map((element) => getTextFromHtml(element?.text)).join('') ?? '';
}

**Comment by greg-schroeder** [comment]
cc @nkdengineer

**Comment by nkdengineer** [comment]
checking this now.

**Comment by nkdengineer** [comment]
cc @c3024 The PR is ready.

**Comment by deetergp** [comment]
This should be fixed now. I have asked for a retest [here](https://expensify.slack.com/archives/C9YU7BX5M/p1719329412981739).

**Comment by kavimuru** [comment]
@deetergp issue fixed

https://github.com/Expensify/App/assets/43996225/50b57465-88df-4356-83d6-f6d0780f6605

**Comment by deetergp** [comment]
Since the bug is fixed I'm going to close this one out. Thanks all 👋
